dotnet/src/SemanticKernel.Core/PromptTemplate/KernelPromptTemplate.cs

```diff
@@ -39,6 +42,9 @@ public KernelPromptTemplate(PromptTemplateConfig promptConfig, ILoggerFactory? l
 
         this._blocks = this.ExtractBlocks(promptConfig, loggerFactory);
         AddMissingInputVariables(this._blocks, promptConfig);
+
+        this._allowUnsafeContent = allowUnsafeContent || promptConfig.AllowUnsafeContent;
+        this._safeBlocks = promptConfig.InputVariables.Where(iv => allowUnsafeContent || iv.AllowUnsafeContent).Select(iv => iv.Name).ToList();
```

Comment:
Small optimization: You could avoid iterating the list if `allowUnsafeContent` is false and just set _safeBlocks to an empty list.

Reply:
If `allowUnsafeContent` is false there is still the case where `iv.AllowUnsafeContent` is true

I will change to using a HashSet